File: parent-337/pinyougou-web-manager/src/main/webapp/js/service/typeTemplateService.js
app.service('typeTemplateService', function ($http) {
    // 分页+条件查询
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../typeTemplate/search.do?pageNum=' + pageNum + '&pageSize=' + pageSize, searchEntity);
    };

    // 新增模板的保存模板
    this.add = function (entity) {
        return $http.post('../typeTemplate/add.do', entity);
    };

    // 修改模板之数据回显
    this.findOne = function (id) {
        return $http.get('../typeTemplate/findOne.do?id=' + id);
    };

    // 修改模板
    this.update = function (entity) {
        return $http.post('../typeTemplate/update.do', entity);
    };

    // 删除模板
    this.del = function (selectIds) {
        return $http.get('../typeTemplate/del.do?selectIds=' + selectIds);
    };

    this.findTypeTemplateList = function () {
        return $http.get('../typeTemplate/findTypeTemplateList.do');
    }
});

File: parent-337/pinyougou-interface/src/main/java/cn/itcast/core/service/TypeTemplateService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.template.TypeTemplate;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface TypeTemplateService {
    /**
     * 分页+条件查询
     *
     * @param pageNum
     * @param pageSize
     * @param typeTemplate
     * @return
     */
    PageResult search(Integer pageNum, Integer pageSize, TypeTemplate typeTemplate);


    /**
     * 新增模板
     *
     * @param typeTemplate
     */
    void add(TypeTemplate typeTemplate);

    /**
     * 修改模板之数据回显
     *
     * @param id
     * @return
     */
    TypeTemplate findOne(Long id);


    /**
     * 修改模板
     *
     * @param typeTemplate
     */
    void update(TypeTemplate typeTemplate);

    /**
     * 删除模板
     *
     * @param selectIds
     * @return
     */
    void del(Long[] selectIds);

    /**
     * 模板下拉框
     */
    List<Map> findTypeTemplateList();

    /**
     * 自定义根据id查询  返回值为List<Map>
     */
    public List<Map> findSpecListById(Long id);


}
